core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -181,11 +181,13 @@ object Controller {
         val instance = if (args.length > 0) args(1).toInt else 0
 
         // initialize the runtimes manifest
-        if (config.isValid && ExecManifest.initialize(config)) {
+        val execManifest = ExecManifest.initialize(config)
+        if (config.isValid && execManifest.isSuccess) {
             val port = config.servicePort.toInt
             BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
         } else {
-            logger.error(this, "Bad configuration, cannot start.")
+            logger.error(this, "Bad configuration, cannot start." +
+              (if (execManifest.isFailure) " Invalid runtimes.manifest: " + execManifest.failed.get.toString else ""))
```

Comment:
how about making this a separate log message?

```
-        // initialize the runtimes manifest
-        val execManifest = ExecManifest.initialize(config)
-        if (config.isValid && execManifest.isSuccess) {
-            val port = config.servicePort.toInt
-            BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
-        } else {
-            logger.error(this, "Bad configuration, cannot start." +
-              (if (execManifest.isFailure) " Invalid runtimes.manifest: " + execManifest.failed.get.toString else ""))
+        def abort() = {
+            logger.error(this, "Bad configuration, cannot start.")
             actorSystem.terminate()
             Await.result(actorSystem.whenTerminated, 30.seconds)
         }
+
+        // initialize the runtimes manifest
+        if (config.isValid) {
+            val execManifest = ExecManifest.initialize(config)
+            if (execManifest.isSuccess) {
+                val port = config.servicePort.toInt
+                BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
+            } else {
+                logger.error(this, s"Invalid runtimes manifest: ${execManifest.failed.get}")
+                abort()
+            }
+        } else abort()
```

Note that pulling the exec manifest initialization to precede the configuration check changes the behavior on which error is caught/reported first.